### PR TITLE
Add a Migration section

### PR DESCRIPTION
**Pull request by jrobinAV**
(cherry picked from commit a4e7496d1d2d0730da6b18aeeafa5f7fdfb90440)

### CODE REVIEW

File: mkdocs.yml_template [hunk 1]
```diff
@@ -99,7 +99,7 @@ nav:
     - "GUI Extension API":
       - manuals/reference_guiext/index.md
       - "taipy-gui module": manuals/reference_guiext/modules.md
-  - "Upgrades": 'manuals/legacy-version-migration.md'
+  - "Upgrades": 'legacy-version-migration.md'
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]: Actually I would not give access to this page from the very top.
New users will not start by upgrading or migrating.
Legacy users will know that the Release Notes will help.

And room in the menu bar is precious...

File: docs/legacy-version-migration.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+---
```
Comment (FabienLelaquais) [inline docs/legacy-version-migration.md]: I would have called this file... 'migration'. We're certainly talking about migrating from a legacy version.

File: docs/legacy-version-migration.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+## From 2.0 to 2.1
+
+In Taipy version 2.1, the version management system has been introduced. For
+applications created with a Taipy Core version &#8804 2.0, the first time it
+runs with version 2.1 or later, no version exists, and so legacy entities are not
+attached to any version. Depending on the mode used to run the application,
+(Refer to [versioning documentation](manuals/core/versioning/index.md)
+for details) we propose the following migration paths:
+
+The overall principle is to create a version the first time the application
```
Comment (FabienLelaquais) [inline docs/legacy-version-migration.md]: May be you want to switch these two last lines, so the semicolon introduces the sub-sections.

File: docs/legacy-version-migration.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+runs with Taipy 2.1 or later and to assign all the old entities to this version.
+
+### Using default or development mode
+
+Please refer to [Development mode](manuals/core/versioning/development_mode.md)
+documentation page for more details on how to run Taipy on development mode.
```
Comment (FabienLelaquais) [inline docs/legacy-version-migration.md]: "on development mode"
->
"in development mode"?

File: docs/legacy-version-migration.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+The first time you run the application with Taipy 2.1 or later, if you use the
+_development_ mode which is the default mode, Taipy automatically creates an
+_experiment_ version with the current configuration and assign all legacy
+entities to it. The version is named "LEGACY-VERSION". Depending on how you
+want to handle legacy entities, you can now manage your newly created version
+using the version management system. Please refer to the
+[Version management system](manuals/core/versioning/index.md) documentation page
+for more details.
+
+### Using Experiment or production mode
```
Comment (FabienLelaquais) [inline docs/legacy-version-migration.md]: Capital "Experiment", lower-case "production".
Any reason for the discrepancy?

File: docs/relnotes.md [hunk 6]
```diff
@@ -6,6 +6,8 @@ hide:
 # Release Notes
 
 This is the list of changes to Taipy releases as they were published.
+Please refer to [upgrades](./legacy-version-migration.md) page for
```
Comment (FabienLelaquais) [inline docs/relnotes.md]: I would not use 'upgrade' here, which would be used when moving from any X.Y to Z.T (Z.T > X.Y).
"migration" makes more sense to me.

File: docs/relnotes.md [hunk 7]
```diff
@@ -17,6 +19,8 @@ Published on 2023-01.
 [`taipy-core` 2.1](https://pypi.org/project/taipy-core/2.1.0/) and
 [`taipy-rest` 2.1](https://pypi.org/project/taipy-rest/2.1.0/) packages.
 
+Please refer to [upgrades](./legacy-version-migration.md#from-2.0-to-2.1) page for
```
Comment (FabienLelaquais) [inline docs/relnotes.md]: Same remark here.

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Let's discuss my remarks, if you wish.
Thx!

File: docs/relnotes.md [hunk 8]
```diff
@@ -17,6 +19,8 @@ Published on 2023-01.
...
+details on how to migrate from version &#8804 2.0 to 2.1.
```
Comment (trgiangdo) [inline docs/relnotes.md]:
```suggestion
details on how to migrate your legacy Taipy application from version &#8804 2.0 to 2.1.
```
The page is to help user migrate an application from 2.0 -> 2.1. If they start a new application, they would not need to read this.

File: docs/relnotes.md [hunk 9]
```diff
@@ -6,6 +6,8 @@ hide:
 # Release Notes
 
 This is the list of changes to Taipy releases as they were published.
+Please refer to [upgrades](./legacy-version-migration.md) page for
+details on how to upgrade Taipy versions.
```
Comment (trgiangdo) [inline docs/relnotes.md]:
```suggestion
details on how to upgrade Taipy version.
```
No `s` right?

**Review by trgiangdo — COMMENTED** [review]
Minor remarks on the release notes

File: docs/relnotes.md (hunk 8 above)
Comment (FabienLelaquais) [reply docs/relnotes.md]: Hopefully... the page is for showing how to migrate from any version to the next, if necessary.
Just like the Release Notes, we don't want to have one page for each new version.

File: docs/relnotes.md (hunk 9 above)
Comment (FabienLelaquais) [reply docs/relnotes.md]: "Upgrading" a Python module is not something we want to document. That would be pip.
"Migrating an application", on the other hand, certainly is one of our concerns.

I would phrase that: "how to migrate your application from 2.0 to 2.1"
and move that reference to the Release Notes for 2.1 only.

File: docs/legacy-version-migration.md (hunk 5 above)
Comment (jrobinAV) [reply docs/legacy-version-migration.md]: Just a typo. Thx.

File: mkdocs.yml_template (hunk 1 above)
Comment (jrobinAV) [reply mkdocs.yml_template]: "Lopin com pru.". I don't understand what you would like.

Comment (FabienLelaquais) [reply mkdocs.yml_template]: We're good now.

**Review by FabienLelaquais — APPROVED** [review]
👌

File: docs/relnotes.md (hunk 9 above)
Comment (FabienLelaquais) [reply docs/relnotes.md]: Done now

File: docs/relnotes.md (hunk 8 above)
Comment (FabienLelaquais) [reply docs/relnotes.md]: Done.

**Pull request merged**